### DISCUSSION TITLE
Is it possible to filter the collision between two soft body's specific tetrahedrons?

### DISCUSSION THREAD

**Discussion by yafei-ou** [NVIDIA-Omniverse/PhysX#234] [Q&A] [answered]
Hello,

I'm wondering if it is possible to filter the collision between two soft body's tetrahedrons. For instance, I would like some tetrahedrons in one soft body's collision mesh to not collide with some other tetrahedrons in another soft body, while the other tetrahedrons are intact. Is this generally achievable?

I noticed that there is this method in PxSoftBody, but I don't have any idea how it can be used.

```
virtual void addSoftBodyFilter(PxSoftBody *otherSoftBody, PxU32 otherTetIdx, PxU32 tetIdx1) = 0
Creates collision filter between a tetrahedron in a soft body and a tetrahedron in another soft body.
```

Thanks for the help in advance.

**Comment by sschirm** [comment, marked as answer]
Hi @yafei-ou , 
Yes, the method you point out is the way to go. If you don't want to have tet, a, b, c of softbody A to not collide with tet d, e in softbody B, then you need to call all combinations.
A->addSoftBodyFilter(B, d, a)
A->addSoftBodyFilter(B, d, c)
A->addSoftBodyFilter(B, d, c)
A->addSoftBodyFilter(B, e, a)
A->addSoftBodyFilter(B, e, c)
A->addSoftBodyFilter(B, e, c)
There is also a batched version: addSoftBodyFilters
Maybe we'll support some grouping feature in the future, but no solid plan yet.
You can also use PX_MAX_TETID for the tetId to say that one tet should be filtered against all other that's of the other softbody (which is usually good enough...)

**Reply by yafei-ou** [reply]
This works, thanks so much for the help.
